doc/man7/migration_guide.pod

```diff
@@ -0,0 +1,1814 @@
+
+=item EVP_PKEY_encrypt_old, EVP_PKEY_decrypt_old, 
+
+Applications should use L<EVP_PKEY_encrypt_init(3)> and L<EVP_PKEY_encrypt(3)> or
+L<EVP_PKEY_decrypt_init(3)> and L<EVP_PKEY_decrypt(3)> instead.
+
+=item EVP_PKEY_get0
+
+This function returns NULL if the key comes from a provider.
+
+=item EVP_PKEY_get0_DH, EVP_PKEY_get0_DSA, EVP_PKEY_get0_EC_KEY, EVP_PKEY_get0_RSA
+=item EVP_PKEY_get1_DH, EVP_PKEY_get1_DSA, EVP_PKEY_get1_EC_KEY and EVP_PKEY_get1_RSA
+=item EVP_PKEY_get0_hmac, EVP_PKEY_get0_poly1305 and EVP_PKEY_get0_siphash
+
+See L</Functions that return an internal key should be treated as read only>.
+
+=item EVP_PKEY_meth_*
+
+See L</Providers are a replacement for engines and low-level method overrides>.
+
+=item EVP_PKEY_new_CMAC_key
+
+See L</Deprecated low-level MAC functions>.
+
+=item EVP_PKEY_assign, EVP_PKEY_set1_DH, EVP_PKEY_set1_DSA, EVP_PKEY_set1_EC_KEY and EVP_PKEY_set1_RSA
+
+See L</Deprecated low-level key object getters and setters>
+
+=item EVP_PKEY_set1_tls_encodedpoint() and EVP_PKEY_get1_tls_encodedpoint().
+
+These functions were previously used by libssl to set or get an encoded public
+key into/from an EVP_PKEY object. With OpenSSL 3.0 these are replaced by the more
+generic functions L<EVP_PKEY_set1_encoded_public_key(3)> and
+L<EVP_PKEY_get1_encoded_public_key(3)>.
+The old versions have been converted to deprecated macros that just call the
+new functions.
+
+=item EVP_PKEY_set1_engine and EVP_PKEY_get0_engine
+
+See L</Providers are a replacement for engines and low-level method overrides>.
+
+=item EVP_PKEY_set_alias_type
+
+This function has been removed. There is no replacement.
+See L</EVP_PKEY_set_alias_type() method has been removed>
+
+=item HMAC_Init_ex, HMAC_Update, HMAC_Final, HMAC_size
+
+=item HMAC_CTX_new, HMAC_CTX_free, HMAC_CTX_copy, HMAC_CTX_reset,
+HMAC_CTX_set_flags and HMAC_CTX_get_md
+
+See L</Deprecated low-level MAC functions>.
+
+=item i2d_DHparams, i2d_DHxparams
+
+=item i2d_DSAparams, i2d_DSAPrivateKey, i2d_DSAPrivateKey_bio,
+i2d_DSAPrivateKey_fp, i2d_DSA_PUBKEY, i2d_DSA_PUBKEY_bio, i2d_DSA_PUBKEY_fp and
+i2d_DSAPublicKey
+
+=item i2d_ECParameters, i2d_ECPrivateKey, i2d_ECPrivateKey_bio,
+i2d_ECPrivateKey_fp, i2d_EC_PUBKEY, i2d_EC_PUBKEY_bio, i2d_EC_PUBKEY_fp and
+i2o_ECPublicKey.
+
+=item i2d_RSAPrivateKey, i2d_RSAPrivateKey_bio, i2d_RSAPrivateKey_fp.
+i2d_RSA_PUBKEY, i2d_RSA_PUBKEY_bio, i2d_RSA_PUBKEY_fp, i2d_RSAPublicKey
+i2d_RSAPublicKey_bio and i2d_RSAPublicKey_fp
+
+See L</Deprecated low-level key reading and writing functions>
+and L<d2i_RSAPrivateKey(3)/Migration> 
+
+=item IDEA_encrypt, IDEA_set_decrypt_key, IDEA_set_encrypt_key,
+IDEA_cbc_encrypt, IDEA_cfb64_encrypt, IDEA_ecb_encrypt and IDEA_ofb64_encrypt
+
+=item MD2, MD2_Init, MD2_Update and MD2_Final
+
+=item MD4, MD4_Init, MD4_Update, MD4_Final and MD4_Transform
+
+=item MDC2, MDC2_Init, MDC2_Update and MDC2_Final
+
+See L</Deprecated low-level encryption functions>.
```

Comment:
Again consistency.  It would be okay to merge all the MD calls into one item and just have the text twice.